File: java/Easy/ConcatenationOfArray.java
import java.util.Arrays;

public class ConcatenationOfArray {
    public static void main(String[] args) {
        int[] nums = {1, 2, 1};
        System.out.println(Arrays.toString(getConcatenation(nums)));
    }
    public static int[] getConcatenation(int[] nums) {
        int[] nums2 = new int[2*nums.length];
        for (int i = 0; i<nums.length;i++){
            nums2[i]=nums[i];
            nums2[i+nums.length]=nums[i];
        }
        return nums2;
    }
    
}

File: kotlin/Easy/ConcatenationOfArray.kt
package Easy

import java.util.*

fun main (){
    val a : IntArray = intArrayOf(1, 2, 3, 1)
    println(Arrays.toString(getConcatenation(a)))
}

fun getConcatenation(nums: IntArray): IntArray {
    val a = nums.size
    val nA = IntArray(a * 2)
    for(i in nums.indices) {
        nA[i] = nums[i]
        nA[i + a] = nums[i]
    }
    return nA
}